api/src/school/domain/usecases/get-school-by-code.js

```diff
@@ -1,7 +1,9 @@
+import { mapToOrganizationLearnerDtos } from '../mappers/map-to-organization-learner-dtos.js';
+
 const getSchoolByCode = async function ({ code, schoolRepository, organizationLearnerRepository } = {}) {
   const school = await schoolRepository.getByCode(code);
   const students = await organizationLearnerRepository.getStudentsByOrganizationId(school.id);
-  school['organizationLearners'] = students;
+  school['organizationLearners'] = mapToOrganizationLearnerDtos(students);
```

Comment:
Proposition : 
On crée un modèle `School` que l'on instancie à partir du tableau de `students` et de l'objet `school` retourné par le `schoolRepository`.
```suggestion
  return new School({ ...school, students });
```

On déplace la logique de rédaction d'éditions des noms de famille dans le getter de `school.organizationLearners`.

Reply:
Je trouve ça cool 👍

Reply:
j'ai fait une proposition dans le commit [`2e261e0` (#8347)](https://github.com/1024pix/pix/pull/8347/commits/2e261e0acde8e9d315602d5ffb905e9ac536c5cd)

Reply:
J'ai fait les modifs qui manquaient. Si c'est ok comme ça, on part là dessus :)